src/components/GameRound.vue

```diff
@@ -30,6 +31,12 @@ const handleEndRoundClick = () => {
   }
 };
 
+let isResetGameModalHidden = ref<boolean>(true);
+
+watch(router.currentRoute, () => {
+  isResetGameModalHidden.value = true;
+});
```

Comment:
I don't see the point of having the modal globaly. Why it can't be a simple component that is depended on the local state and rendered internally?